File: apps/social/test/lib/envoy-ai-slash-commands.test.ts
import { describe, expect, it } from "vitest";
import {
  formatEnvoyAiSlashHelp,
  parseEnvoyAiSlashCommand,
} from "../../src/lib/envoy-ai-slash-commands.js";
import type { ExtAgentCommandCatalog } from "@envoymesh/api";

describe("envoy-ai-slash-commands", () => {
  it("parses envoy-owned actions", () => {
    expect(parseEnvoyAiSlashCommand("/help")).toEqual({ type: "help" });
    expect(parseEnvoyAiSlashCommand("/clear")).toEqual({ type: "clear" });
    expect(parseEnvoyAiSlashCommand("/status")).toEqual({ type: "status" });
    expect(parseEnvoyAiSlashCommand("/skills")).toEqual({ type: "skills" });
    expect(parseEnvoyAiSlashCommand("/report")).toEqual({ type: "report" });
  });

  it("expands hybrid mesh and feature prompts", () => {
    expect(parseEnvoyAiSlashCommand("/bonds")).toMatchObject({ type: "expand" });
    expect(parseEnvoyAiSlashCommand("/knowledge parking meters")).toMatchObject({
      type: "expand",
      prompt: expect.stringContaining("parking meters"),
    });
    expect(parseEnvoyAiSlashCommand("/knowledge")).toEqual({ type: "help" });
    expect(parseEnvoyAiSlashCommand("/about")).toMatchObject({
      type: "expand",
      prompt: expect.stringContaining("EnvoyMesh"),
    });
    expect(parseEnvoyAiSlashCommand("/terminal")).toMatchObject({
      type: "expand",
      prompt: expect.stringContaining("Terminals"),
    });
    expect(parseEnvoyAiSlashCommand("/team")).toMatchObject({
      type: "expand",
      prompt: expect.stringContaining("Office LAN"),
    });
    expect(parseEnvoyAiSlashCommand("/team")?.type === "expand" &&
      parseEnvoyAiSlashCommand("/team")!.prompt).toContain("Manage workers");
    expect(parseEnvoyAiSlashCommand("/family")).toMatchObject({ type: "expand" });
    expect(parseEnvoyAiSlashCommand("/extagent")).toMatchObject({ type: "expand" });
    expect(parseEnvoyAiSlashCommand("/envoyai")).toMatchObject({ type: "expand" });
    expect(parseEnvoyAiSlashCommand("/pi")).toMatchObject({ type: "expand" });
    expect(parseEnvoyAiSlashCommand("/content")).toMatchObject({
      type: "expand",
      prompt: expect.stringContaining("Feed"),
    });
  });

  it("formats help from catalog", () => {
    const catalog: ExtAgentCommandCatalog = {
      agentId: "envoyai",
      agentName: "EnvoyAI",
      commands: [
        {
          slash: "/help",
          summary: "List commands",
          intercept: "envoy",
          source: "static",
        },
      ],
      catalogVersion: "2",
      fetchedAt: "2026-08-12T00:00:00.000Z",
      limitations: ["Note"],
    };
    const text = formatEnvoyAiSlashHelp(catalog);
    expect(text).toContain("/help");
    expect(text).toContain("• Note");
  });
});

File: apps/envoygo/test/ext_agent/envoy_ai_slash_commands_test.dart
import 'package:envoygo/ext_agent/envoy_ai_slash_commands.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('parses EnvoyAI slash actions', () {
    expect(parseEnvoyAiSlashCommand('/help')?.type, 'help');
    expect(parseEnvoyAiSlashCommand('/clear')?.type, 'clear');
    expect(parseEnvoyAiSlashCommand('/bonds')?.type, 'expand');
    expect(
      parseEnvoyAiSlashCommand('/knowledge parking')?.prompt,
      contains('parking'),
    );
  });

  test('expands feature guide prompts', () {
    expect(parseEnvoyAiSlashCommand('/about')?.type, 'expand');
    expect(parseEnvoyAiSlashCommand('/about')?.prompt, contains('EnvoyMesh'));
    expect(parseEnvoyAiSlashCommand('/terminal')?.prompt, contains('Terminals'));
    expect(parseEnvoyAiSlashCommand('/team')?.type, 'expand');
    expect(parseEnvoyAiSlashCommand('/team')?.prompt, contains('Office LAN'));
    expect(parseEnvoyAiSlashCommand('/team')?.prompt, contains('Manage workers'));
    expect(parseEnvoyAiSlashCommand('/team')?.prompt, contains('fleet token'));
    expect(parseEnvoyAiSlashCommand('/family')?.type, 'expand');
    expect(parseEnvoyAiSlashCommand('/extagent')?.type, 'expand');
    expect(parseEnvoyAiSlashCommand('/envoyai')?.type, 'expand');
    expect(parseEnvoyAiSlashCommand('/pi')?.type, 'expand');
    expect(parseEnvoyAiSlashCommand('/content')?.prompt, contains('Feed'));
  });

  test('formats help', () {
    final text = formatEnvoyAiSlashHelp({
      'commands': [
        {'slash': '/help', 'summary': 'List commands'},
        {'slash': '/report', 'summary': 'Mesh report'},
        {'slash': '/about', 'summary': 'Product overview'},
      ],
      'limitations': ['Note'],
    });
    expect(text, contains('EnvoyAI slash commands:'));
    expect(text, contains('/report'));
    expect(text, contains('/about'));
    expect(text, contains('• Note'));
  });
}
